src/server/procedures/myForestV2/contributions.ts

```diff
@@ -326,6 +326,41 @@ function populateContributedCountries(
   if (contributedCountry) contributedCountries.add(contributedCountry);
 }
 
+function getLatestPlantDate(value: MyContributionsMapItem): Date {
+  if (
+    value.type === 'project' &&
+    value.latestContributions &&
+    value.latestContributions.length > 0
+  ) {
+    return new Date(value.latestContributions[0].plantDate);
+  } else if (
+    value.type === 'registration' &&
+    value.contributions &&
+    value.contributions.length > 0
+  ) {
+    return new Date(value.contributions[0].plantDate);
+  } else {
+    return new Date(0); // Return earliest possible date if no plantDate found to make items without a date appear last
+  }
+}
```

Comment:
It can be refactored for improved readability and maintainability.  

- Extract common logic into a separate function.
- Use a constant for the default date to make it clear.
- you can use switch statement

Reply:
1. There isn't any common logic within `getLatestPlantDate` that needs to be abstracted away
2. Done
3. As we're not comparing discrete values, I prefer to use `if else`

Reply:
Done in [45a5c1a](https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/2122/commits/45a5c1a24e47e4f36b63be1c2d7f632e84f8a7c8)